### PR TITLE
.github/workflows: run integration tests in parallel matrix mode

### PR DESCRIPTION
**Pull request by paulfantom**
## Description

~- adding helm chart tests~ I'll add it in a separate PR due to amount of work needed to be done.
- moving renaming some e2e tests to integration tests
- making tests run in parallel.
- shortening or removing test names to allow better visibility into status checks on PR page

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

```release-note

```

### CODE REVIEW

File: ct.yaml
Comment (JamesGuthrie) [inline ct.yaml:5]: I think that this is supposed to be `helm-chart`. When I set that, `ct` complains about the directory not being a valid chart directory:

> Directory 'helm-chart/templates' is not a valid chart directory. Skipping...

File: .github/workflows/go.yml
Comment (JamesGuthrie) [inline .github/workflows/go.yml:148]: It seems as though the "Go End-to-End Test Results" job could also be renamed in the same pattern (to "e2e results" perhaps)?

**Review by JamesGuthrie — CHANGES_REQUESTED** [review]

File: .github/workflows/go.yml
Comment (paulfantom) [reply .github/workflows/go.yml:148]: Makes sense.

@cevian repo owner will need to amend branch protection options to mark new check name as required one.

File: ct.yaml
Comment (paulfantom) [reply ct.yaml]: Good catch! :+1:

Comment (JamesGuthrie) [reply ct.yaml]: This still doesn't work for me. I also tried running it in a docker container. Am I missing something?

**Review by cevian — APPROVED** [review]
LGTM - very nice improvements. Didn't know about the ct tool

**Review by JamesGuthrie — CHANGES_REQUESTED** [review]
I took another look at this and I'm still not sure that it's actually working. Here are my steps to reproduce:

```
# enter promscale repo
> cd promscale
> docker run --platform=linux/amd64 -v$(pwd):/files -ti quay.io/helmpack/chart-testing bash

# remaining commands run inside docker container

> cp /files/helm-chart/Chart.yaml /tmp/Chart.yaml.bak

# make an invalid change to the chart
> echo 'type: foo' >> /files/helm-chart/Chart.yaml
> cd /files
> ct lint --config ct.yaml

Linting charts...
Directory 'helm-chart' is not a valid chart directory. Skipping...
------------------------------------------------------------------------------------------------------------------------
No chart changes detected.
------------------------------------------------------------------------------------------------------------------------
All charts linted successfully

> cd helm-chart
> helm lint

==> Linting .
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/: validation: chart.metadata.type must be application or library
[ERROR] : unable to load chart
	validation: chart.metadata.type must be application or library

Error: 1 chart(s) linted, 1 chart(s) failed

> cp  /tmp/Chart.yaml.bak /files/helm-chart/Chart.yaml
```

**Comment by paulfantom** [comment]
Seems that I used ct version 2.5.0 and CI is using 3.4.0 :facepalm: I'll refactor the PR. Thanks @JamesGuthrie for catching this!

**Comment by paulfantom** [comment]
I've removed adding helm chart tests and will add them in a separate PR due to amount of changes needed to be done. This PR is now limited only to simplifying existsing tests.

**Review by JamesGuthrie — CHANGES_REQUESTED** [review]
Can you rewrite the history to remove the commit which adds ct?

**Comment by paulfantom** [comment]
@JamesGuthrie Sure, done ;)

**Review by JamesGuthrie — APPROVED** [review]
LGTM

**Comment by paulfantom** [comment]
bors merge

**Comment by paulfantom** [comment]
> Timed out.

FFS. -.-

bors merge

**Comment by paulfantom** [comment]
bors merge

**Comment by paulfantom** [comment]
bors merge

**Comment by JamesGuthrie** [comment]
bors ping

**Comment by JamesGuthrie** [comment]
bors merge

**Comment by JamesGuthrie** [comment]
Bors will never allow this change to be merged. In changing the test names we broke the bors config. The following should fix it:

```
diff --git a/.github/workflows/integration.yml b/.github/workflows/integration.yml
index d43917f..ee8ac24 100644
--- a/.github/workflows/integration.yml
+++ b/.github/workflows/integration.yml
@@ -7,7 +7,7 @@ on:
     branches: [master, main, force_test, release-*]
 
 jobs:
-  run:
+  test-integration:
     runs-on: ubuntu-latest
     strategy:
       matrix:
@@ -33,3 +33,19 @@ jobs:
     - name: ${{ matrix.test-setups.name }}
       run: ./scripts/end_to_end_tests.sh ${{ matrix.test-setups.image }}
       shell: bash
+
+  # Added to summarize the matrix (otherwise we would need to list every single
+  # job in bors.toml)
+  run-result:
+    name: integration results
+    if: always()
+    needs:
+      - test-integration
+    runs-on: ubuntu-latest
+    steps:
+      - name: Mark the job as a success
+        if: needs.test-integration.result == 'success'
+        run: exit 0
+      - name: Mark the job as a failure
+        if: needs.test-integration.result != 'success'
+        run: exit 1
diff --git a/bors.toml b/bors.toml
index a51e0c0..b802442 100644
--- a/bors.toml
+++ b/bors.toml
@@ -1,4 +1,4 @@
-status = ["Build and Lint", "Go Tests", "Go End-to-End Test Results", "Bash End-to-End Tests"]
+status = ["Build and Lint", "unit", "e2e results", "integration results"]
 delete-merged-branches = true
 required_approvals = 1
 timeout_sec = 3600 # one hours
```

**Comment by paulfantom** [comment]
Fortunately we are deprecating bors :)

@cevian this is blocked until bors is switched off

File: .github/workflows/integration.yml
Comment (cevian) [inline .github/workflows/integration.yml:35]: do we need the equivalent of "e2e results" here to sum up the results so we can put it into branch protection rules

**Comment by cevian** [comment]
Note: we need to fix the branch protection rules after this is merged. @paulfantom please remind me.

File: .github/workflows/integration.yml
Comment (JamesGuthrie) [reply .github/workflows/integration.yml:35]: I think so.

Comment (paulfantom) [reply .github/workflows/integration.yml:35]: added

**Comment by paulfantom** [comment]
I've added an integration test summary job.

@cevian I cannot merge this because of branch protection rules :)

**Comment by JamesGuthrie** [comment]
@paulfantom can you try merging now?

**Comment by paulfantom** [comment]
Yes, done. Thanks @JamesGuthrie :+1:

**Comment by JamesGuthrie** [comment]
@cevian I have adjusted the branch protection rules to account for the renamings.

**Pull request merged**
